src/main.ts

```diff
@@ -196,24 +313,19 @@ async function drawFrame(index: number): Promise<void> {
 }
 
 async function drawLoop(): Promise<void> {
-  drawLoopRunning = true;
-  const oneArrowDown = (leftArrowDown || rightArrowDown) && !(leftArrowDown && rightArrowDown);
-  if (dataset && datasetOpen && oneArrowDown) {
-    await drawFrame(currentFrame);
-    const delta = leftArrowDown ? -1 : 1;
-    currentFrame = (currentFrame + delta + dataset.numberOfFrames) % dataset.numberOfFrames;
+  if (dataset && datasetOpen) {
     // update higlighted cell id if any
     if (selectedTrack) {
-      const id = selectedTrack.getIdAtTime(currentFrame);
+      const id = selectedTrack.getIdAtTime(timeControls.currentFrame);
       canv.setHighlightedId(id - 1);
       // console.log(`selected track: ${selectedTrack.trackId}; highlighted id ${id}`);
     }
     // update current time in plot
-    plot.setTime(currentFrame);
-
+    plot.setTime(timeControls.currentFrame);
+    await drawFrame(timeControls.currentFrame);
+  }
+  if (drawLoopRunning) {
     window.requestAnimationFrame(drawLoop);
-  } else {
-    drawLoopRunning = false;
   }
```

Comment:
Looks like with `TimeControls` now handling playback, `drawLoopRunning` is set to false and never changed, so it and this if statement can go away